docs/architecture/overriding-system-calls.md

```diff
@@ -0,0 +1,27 @@
+# Overriding System Calls
```

Comment:
After thinking about this, we really should have a section with more information on system calls in general.

Let's change the title/filename to `System calls` (note case)